sql-plugin/src/main/scala/com/nvidia/spark/rapids/RapidsMeta.scala

```diff
@@ -554,6 +554,16 @@ abstract class SparkPlanMeta[INPUT <: SparkPlan](plan: INPUT,
     childPlans.foreach(_.tagForExplain())
   }
 
+  def requireAstForGpuOn(exprMeta: BaseExprMeta[_]): Unit = {
+    val previousExprReplaceVal = canExprTreeBeReplaced
+    exprMeta.requireAstForGpu()
+    val newExprReplaceVal = canExprTreeBeReplaced
```

Comment:
This code is a little confusing, I think a comment would help.  I'm assuming this is covering the case where something can be translated to an AST but not translated in the general sense?  Otherwise I would expect this to just call `canExprTreeBeReplaced` after requiring AST.